### PR TITLE
fixed bugs for registration, booking and payment

### PR DESCRIPTION
**Pull request by ghost**
Fixed bugs:

31/5/23
-clicking on booking time redirects to home page. 
-card number allows non-numeric characters. 
-successful registration does not redirect to home page. 
-does not check valid email.

### CODE REVIEW

File: bugs.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+-card number allows non-numeric characters when at length 12.
```
Comment (shirleymn) [inline bugs.txt]: Yeah so for this file, remove it from your next commit (e.g. move it to your desktop).

1. Move bugs.txt to some location outside of git directory
2. `git status` see what files can be added to staging area
3. `git add .`
4. `git status` see what files have been put into the staging area
5. `git commit -m "your commit message"`
6. `git push`

File: frontend/src/features/checkout/Checkout.js [hunk 2]
```diff
@@ -18,27 +18,40 @@ const Checkout = () => {
...
   };
 
+  // send payment request
   const payment = (e) => {
     e.preventDefault();
     dispatch(resetPayment());
-    dispatch(confirmPayment(order));
+    setError({ status: false, message: "" });
+
+    if (order.card.length === 12 && /[0-9]/.test(order.card)) {
```
Comment (shirleymn) [inline frontend/src/features/checkout/Checkout.js]: I think the regex for only numerical input is why it's not working

`/[0-9]/` is only checking that there is at least one numeric character in the string (according to ChatGPT)....

https://www.w3resource.com/javascript/form/all-numbers.php#:~:text=To%20get%20a%20string%20contains%20only%20numbers%20(0%2D9)%20we%20use%20a%20regular%20expression%20(/%5E%5B0%2D9%5D%2B%24/)%20which%20allows%20only%20numbers.

File: frontend/src/features/films/Film.js [hunk 3]
```diff
@@ -18,35 +18,45 @@ const Film = () => {
...
     time: "13:00",
   });
+  console.log(tickets);
 
   useEffect(() => {
     const selectedFilm = { film_name: params.film_name };
     dispatch(fetchTickets(selectedFilm));
   }, [dispatch, params]);
 
+  useEffect(() => {
```
Comment (shirleymn) [inline frontend/src/features/films/Film.js:32]: What is this hook doing?

**Review by shirleymn — CHANGES_REQUESTED** [review]
Looks good to me, nice changes. Just one small thing about the card validation regex and understanding how film data is displayed.

File: frontend/src/features/films/Film.js (hunk 3 above)
Comment (ghost) [reply frontend/src/features/films/Film.js:32]: this is for setting the initial state for the ticket

File: frontend/src/features/checkout/Checkout.js (hunk 2 above)
Comment (ghost) [reply frontend/src/features/checkout/Checkout.js]: I've solved it with /^[0-9]+$/

**Review by shirleymn — APPROVED** [review]
LGTM

**Pull request merged**